java/src/main/kotlin/io/spine/validation/java/generate/option/PatternFieldGenerator.kt

```diff
@@ -252,3 +251,52 @@ private fun PatternOption.Modifier.asFlagsMask(): Int {
     }
     return mask
 }
+
+/**
+ * ASCII control characters escaped by Protobuf compiler when parsing string literals.
+ *
+ * Escaping rules may differ from compiler to compiler, so a more reliable approach is
+ * to re-escape strings in accordance to the specs of a particular tool that previously
+ * performed this escaping.
+ *
+ * Note we don't touch question marks because Protobuf compiler actually accepts both `?`
+ * and `\?` as a question mark. So, it is unclear when we should prepend the leading `/`
+ * to restore the original literal.
+ *
+ * Source: [Text Format Langauge Specification | String Literals](https://protobuf.dev/reference/protobuf/textformat-spec/#string).
+ */
+@Suppress("MagicNumber") // ASCII codes.
+private val ProtobufEscape = mapOf(
+    7 to "\\a",
+    8 to "\\b",
+    12 to "\\f",
+    10 to "\\n",
+    13 to "\\r",
+    9 to "\\t",
+    11 to "\\v",
+    92 to "\\\\",
+    39 to "\\'",
+    34 to "\\\""
+)
+
+/**
+ * Undoes [ASCII escaping][ProtobufEscape] that Protobuf compiler performs for string literals,
+ * allowing the passed [value] to be used as a string literal in Java code.
+ *
+ * Consider the following example:
+ *
+ * 1. A string literal is defined in a proto file as the following: `[^\\/]`.
+ * 2. Runtime string representation would be the following: `[^\/]`.
```

Comment:
Please add more steps to this example so that it is absolutely clear what happens if we don't escape the code on our side again.

Right now, this example says nothing about our code generation, and the state of things in the generated code.